### PR TITLE
PSL-126 PSL-127  / Implement pastel-utility update and install

### PR DESCRIPTION
**Pull request by bacnh8**
PSL-126
- [x] Write Makefile to compile in platforms: linux/amd64, windows/amd64, linux/amd64
- [x] Implement local Upload function to do: Stop remote supernode service, Upload binary, Restart service (start again). Validate and fix start/stop function
- [x] Change systemd services to usermode, so no remote-pw is needed
- [ ] Implement update binary - from downloading from default server

PSL-127 
- [x] pastelup start node, supernode and supernode-coldhot: When Ctrl-C caught ask user if they really want to terminate. now it is just terminate pastelup immediately

pastelup start supernode-coldhot 
- [x] After starting local pasteld, ask user if they want to wait for it to sync before searching for the collateral transactions, otherwise it will not find the collateral transactions if it was made in high blocks
- [x] Do not stop local pasteld if it was running before pasteld started
- [x]  Better log messages

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (nguyen202) [inline .vscode/launch.json:2]: please remove this file

File: cmd/update.go [hunk 2]
```diff
@@ -2,61 +2,245 @@ package cmd
...
+		files, err := ioutil.ReadDir(updateConfigFlags.BinComponentPath)
 		if err != nil {
 			return err
 		}
 
-		log.Info("flag-name: ", updateFlag)
+		for _, file := range files {
+			log.WithContext(ctx).Infof("Copying %s to remote host %s", file.Name(), config.PastelExecDir)
+			if err := client.Scp(filepath.Join(updateConfigFlags.BinComponentPath, file.Name()), filepath.Join(config.PastelExecDir, file.Name())); err != nil {
+				return err
```
Comment (nguyen202) [inline cmd/update.go:213]: could you log error here?

File: cmd/update.go [hunk 3]
```diff
@@ -2,61 +2,245 @@ package cmd
...
+			}
+		}
+	} else {
+		log.WithContext(ctx).Infof("Copying file %s to %s at remote host", updateConfigFlags.BinComponentPath, config.PastelExecDir)
+		if err := client.Scp(updateConfigFlags.BinComponentPath, filepath.Join(config.PastelExecDir, fileInfo.Name())); err != nil {
+			return err
```
Comment (nguyen202) [inline cmd/update.go:219]: nit: log error here

File: cmd/update.go [hunk 4]
```diff
@@ -2,61 +2,245 @@ package cmd
...
+		log.WithContext(ctx).WithError(err).Error("Failed to stop Supernode services")
+		return err
+	}
+
+	log.WithContext(ctx).Info("Successfully stop supernode at remote host")
+
+	/* Copy the binary (pastel-cli, pasteld, pastel-cli, rq-service, supernode) from local folder to remote location to overwrite binary */
+	fileInfo, err := os.Stat(updateConfigFlags.BinComponentPath)
+	if err != nil {
+		return err
```
Comment (nguyen202) [inline cmd/update.go:200]: nit: log error here

File: cmd/update.go [hunk 5]
```diff
@@ -2,61 +2,245 @@ package cmd
...
+
+	if len(updateConfigFlags.SSHKey) == 0 {
+		username, password, _ := utils.Credentials(updateConfigFlags.RemoteUser, true)
+		client, err = utils.DialWithPasswd(fmt.Sprintf("%s:%d", updateConfigFlags.RemoteIP, updateConfigFlags.RemotePort), username, password)
+	} else {
+		username, _, _ := utils.Credentials(updateConfigFlags.RemoteUser, false)
+		client, err = utils.DialWithKey(fmt.Sprintf("%s:%d", updateConfigFlags.RemoteIP, updateConfigFlags.RemotePort), username, updateConfigFlags.SSHKey)
+	}
+	if err != nil {
+		return err
```
Comment (nguyen202) [inline cmd/update.go:147]: nit: log error here

**Review by a-ok123 — APPROVED** [review]

**Pull request merged**